app/models/katello/product_content.rb

```diff
@@ -52,11 +52,24 @@ def self.with_valid_subscription(organization)
       where(:product_id => Katello::PoolProduct.where(:pool_id => organization.pools).select(:product_id))
     end
 
-    # used by Katello::Api::V2::RepositorySetsController#index
+    # following 4 methods used by Katello::Api::V2::RepositorySetsController#index
     def repositories
       Katello::Repository.in_default_view.where(:root_id => product.root_repositories.has_url.where(:content_id => content.cp_content_id))
     end
 
+    def root_repositories
```

Comment:
Could we have a better name here since `root_repositories` has its own meaning in Katello already?
How about `all_repositories`, `source_repositories`, `base_repositories`?

Reply:
They _are_ root repositories, though! Maybe `all_root_repositories` to highlight the fact that it's not filtered?

Reply:
These are `Katello::Repository` instead of `Katello::RootRepository`.

Reply:
My bad. `all_repositories` sounds good to me, then.